### PR TITLE
Improve FasterRCNN model quality.

### PR DESCRIPTION
**Pull request by tanzhenyu**
This PR add mores paper specific training logic and improves FasterRCNN model quality / training by:
1) add padding
2) add variance during box encoding and decoding
3) less anchor gen
4) larger image size
5) more aggressive resizing

It makes the training script a little more verbose, which is intentional given our data augmentation layers set do not support native box format and paper specific resizing logic.

### CODE REVIEW

File: keras_cv/bounding_box/converters.py [hunk 1]
```diff
@@ -25,6 +25,25 @@ class RequiresImagesException(Exception):
     pass
 
 
+def clip_boxes(boxes, box_format, image_shape):
```
Comment (ianstenbit) [inline keras_cv/bounding_box/converters.py:28]: keras_cv.bounding_box.clip_to_image should do this, right?
(from bounding_box/utils.py)

Comment (ianstenbit) [reply keras_cv/bounding_box/converters.py:28]: (If not, and this needs to be a separate method, maybe it should also live in `utils.py` instead of `converters.py`?)

File: examples/models/object_detection/faster_rcnn/train.py [hunk 2]
```diff
@@ -25,25 +25,145 @@
...
+
+def get_non_empty_box_indices(boxes):
+    # Selects indices if box height or width is 0.
+    height = boxes[:, 2] - boxes[:, 0]
+    width = boxes[:, 3] - boxes[:, 1]
+    indices = tf.where(tf.logical_and(tf.greater(height, 0), tf.greater(width, 0)))
+    return indices[:, 0]
+
+
+def resize_fn(image, gt_boxes, gt_classes):
```
Comment (ianstenbit) [inline examples/models/object_detection/faster_rcnn/train.py:154]: Can this functionality be achieved with RandomCropAndResize? If not, should we either update that layer or create a new KPL which provides this functionality?

File: keras_cv/bounding_box/converters.py (hunk 1 above)
Comment (tanzhenyu) [reply keras_cv/bounding_box/converters.py:28]: Ideally it does, but that would probably error out given:
"`bounding_boxes` must include a class_id index on the final "
-- as discussed last time, this is a huge tech debt we should get rid of, but I don't want it to block the current development as well.

File: examples/models/object_detection/faster_rcnn/train.py (hunk 2 above)
Comment (tanzhenyu) [reply examples/models/object_detection/faster_rcnn/train.py:154]: After training many classification and detection models, I have came to conclusion that a RandomResizeAndCrop-like behavior is required, but it cannot be achieved with the layers we have today.
What I would really like to see is, once I get to train some segmentation models and achieve SOTA, we can try to discuss what the best RandomResizeAndCrop layer should look like.
Note: Also the layer we have today does not support native box format, the same issue with the above comment (as mentioned in the PR description)

Comment (bhack) [reply examples/models/object_detection/faster_rcnn/train.py:154]: I think that we could re-evaluate if some layers need to have an attached policy:
- https://github.com/keras-team/keras-cv/issues/108#issuecomment-1040626481
- https://github.com/keras-team/keras-cv/issues/108#issuecomment-1043598240

**Comment by tanzhenyu** [comment]
@innat Once you patch this PR, you should get better results.

File: examples/models/object_detection/faster_rcnn/train.py (hunk 2 above)
Comment (tanzhenyu) [reply examples/models/object_detection/faster_rcnn/train.py:154]: The resize is somewhat different than other augmentation layers we have, especially for auto/rand-aug, colorized ops are used extensively, and only 1-2 geometric ops are popular choices

File: keras_cv/bounding_box/converters.py (hunk 1 above)
Comment (ianstenbit) [reply keras_cv/bounding_box/converters.py:28]: Gotcha. If that method doesn't work as intended and it's not simple to update it, I think we should keep this version of it internal to the training script.

Otherwise we have two separate but nearly identical utilities.

File: examples/models/object_detection/faster_rcnn/train.py (hunk 2 above)
Comment (ianstenbit) [reply examples/models/object_detection/faster_rcnn/train.py:154]: sgtm. Maybe let's put a TODO in the code here for now to migrate this to a KPL?

File: keras_cv/bounding_box/converters.py (hunk 1 above)
Comment (tanzhenyu) [reply keras_cv/bounding_box/converters.py]: Done.

File: examples/models/object_detection/faster_rcnn/train.py (hunk 2 above)
Comment (tanzhenyu) [reply examples/models/object_detection/faster_rcnn/train.py:154]: Done.

Comment (bhack) [reply examples/models/object_detection/faster_rcnn/train.py:155]: In the case we will want to re-evaluate some policy in the future, at least for the classification task there was a nice survey out recently:
https://arxiv.org/abs/2206.06544

File: keras_cv/bounding_box/converters.py [hunk 3]
```diff
@@ -25,6 +25,25 @@ class RequiresImagesException(Exception):
     pass
 
 
+def _clip_boxes(boxes, box_format, image_shape):
```
[the remark this answers is not in the document]
Comment (tanzhenyu) [reply keras_cv/bounding_box/converters.py]: Good point. Done.

File: examples/models/object_detection/faster_rcnn/train.py (hunk 2 above)
Comment (bhack) [reply examples/models/object_detection/faster_rcnn/train.py:155]: I don't have the permission level to assign it to you but it is here:
https://github.com/keras-team/keras-cv/issues/900
